File: testing-framework/backend/src/services/requirementContent.ts
import { prisma } from "../prisma.js";

export interface GetFullContentResult {
  fullContent: string;
  attachmentErrors: string[];
}

export async function getFullContent(requirementId: string): Promise<GetFullContentResult> {
  const requirement = await prisma.requirement.findUnique({
    where: { id: requirementId },
    include: { attachments: true },
  });
  if (!requirement) throw new Error("Requirement not found");

  const parts: string[] = [`# ${requirement.title}\n`, requirement.content || ""];
  const attachmentErrors: string[] = [];

  console.log(`[getFullContent] requirementId=${requirementId}, 附件数=${requirement.attachments.length}`);

  for (const a of requirement.attachments) {
    if (a.extractedText?.trim()) {
      parts.push(`\n## 附件: ${a.filename}\n${a.extractedText.trim()}`);
    } else if (a.mimeType?.startsWith("image/")) {
      parts.push(`\n## 附件: ${a.filename} (图片)`);
    } else {
      attachmentErrors.push(a.filename);
    }
  }

  const fullContent = parts.join("\n").trim();
  console.log(`[getFullContent] 完成, fullContent 长度=${fullContent.length}, attachmentErrors=${attachmentErrors.length} 个`);
  return {
    fullContent,
    attachmentErrors,
  };
}

File: testing-framework/backend_python/app/services/requirement_content.py
from dataclasses import dataclass

from sqlalchemy.orm import Session

from app.models import Requirement


@dataclass
class GetFullContentResult:
    full_content: str
    attachment_errors: list[str]


def get_full_content(db: Session, requirement_id: str) -> GetFullContentResult:
    req = db.query(Requirement).filter(Requirement.id == requirement_id).first()
    if not req:
        raise ValueError("Requirement not found")
    parts = [f"# {req.title}\n", req.content or ""]
    errors: list[str] = []
    atts = sorted(req.attachments, key=lambda a: a.id)
    print(f"[getFullContent] requirementId={requirement_id}, 附件数={len(atts)}")
    for a in atts:
        et = (a.extractedText or "").strip()
        if et:
            parts.append(f"\n## 附件: {a.filename}\n{et}")
        elif (a.mimeType or "").startswith("image/"):
            parts.append(f"\n## 附件: {a.filename} (图片)")
        else:
            errors.append(a.filename)
    full = "\n".join(parts).strip()
    print(f"[getFullContent] 完成, fullContent 长度={len(full)}, attachmentErrors={len(errors)}")
    return GetFullContentResult(full_content=full, attachment_errors=errors)
